xbmc/addons/GUIDialogAddonInfo.cpp

```diff
@@ -452,6 +453,35 @@ void CGUIDialogAddonInfo::OnUninstall()
   Close();
 }
 
+namespace
+{
+  class AsyncAddonEnable : private IRunnable
+  {
+  public:
+    explicit AsyncAddonEnable(const std::string& strAddonId) : m_strAddonId(strAddonId) {}
+    bool Execute() { return CGUIDialogBusy::Wait(this, 100, true); }
```

Comment:
CGUIDialogBusy::Wait seems to be implemented incorrectly. If you allow cancel, it just cancels the busy dialog and leaves the action running. That puts the app in an unconsistant state.

Reply:
Thanks for spotting. I actually don't want the action to be cancel-able, because it isn't, regardless of the CGUIDialogBusy bug you mentioned. I will change the respective parameter to false.